### PR TITLE
Add log sink header and spec

### PR DESCRIPTION
**Pull request by dcristoloveanu**

### CODE REVIEW

File: v2/devdoc/log_sink_requirements.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# `log_sink` requirements
+
+`log_sink` defines an interface for a log sink (as a structure that contains one function pointer).
+
+A log sink has the responsibility to output a logging event (be that to the console, producting an ETW event, writing it to a file, etc.)
+
+## Exposed API
+
+```
+typedef void (*LOG_SINK_LOG_FUNC)(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, char* message, const char* file, const char* func, int line);
```
Comment (anporumb) [inline v2/devdoc/log_sink_requirements.md:10]: >char [](http://example.com/codeflow?start=87&length=4)

why not const?

File: v2/inc/c_logging/log_sink.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#define LOG_SINK_H
+
+#include "c_logging/log_context.h"
+
+typedef void (*LOG_SINK_LOG_FUNC)(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, char* message, const char* file, const char* func, int line);
+
+typedef struct LOG_SINK_TAG
+{
+    LOG_SINK_LOG_FUNC log_sink_log;
+} LOG_SINK;
```
Comment (anporumb) [inline v2/inc/c_logging/log_sink.h:14]: >LOG_SINK [](http://example.com/codeflow?start=2&length=8)

Is this a wanna-be interface? [with just 1 function?]

Comment (anporumb) [reply v2/inc/c_logging/log_sink.h:14]: line 3 of the .md says it is. Then consider naming it with `_IF`

**Review by anporumb — APPROVED** [review]
:shipit:

File: v2/devdoc/log_sink_requirements.md (hunk 1 above)
Comment (dcristoloveanu) [reply v2/devdoc/log_sink_requirements.md]: Made const

File: v2/inc/c_logging/log_sink.h (hunk 2 above)
Comment (dcristoloveanu) [reply v2/inc/c_logging/log_sink.h]: Renamed to _IF. As part of that also extracted LOG_LEVEL in its own header.

**Pull request merged**